sphinx_proof/nodes.py

```diff
@@ -25,35 +30,53 @@ class unenumerable_node(nodes.Admonition, nodes.Element):
 
 
 def visit_enumerable_node(self, node: Node) -> None:
-    self.body.append(self.starttag(node, "div", CLASS="admonition"))
+    if isinstance(self, LaTeXTranslator):
+        docname = find_parent(self.builder.env, node, "section")
+        self.body.append("\\label{" + f"{docname}:{node.attributes['label']}" + "}")
+        self.body.append(latex_admonition_start)
+    else:
+        self.body.append(self.starttag(node, "div", CLASS="admonition"))
 
 
 def depart_enumerable_node(self, node: Node) -> None:
     typ = node.attributes.get("type", "")
-
-    # Find index in list of 'Proof #'
-    number = get_node_number(self, node)
-    idx = self.body.index(f"Proof {number} ")
-    self.body[idx] = f"{typ.title()} {number} "
-    self.body.append("</div>")
+    if isinstance(self, LaTeXTranslator):
+        number = get_node_number_latex(self, node)
```

Comment:
@AakashGfude why are the node number functions different between `html` and `latex`?

Reply:
@mmcky because in latex we don't have the variable `self.builder.fignumbers` used in HTML. Basically, this variable gives us a document-level dictionary of proof/algorithm...etc(in our case)  numbers.
We have to manually construct a variable like that using `self.builder.env.toc_fignumbers` for the latex case. Because latex only has one single concatenated document and does not have the notion of individual document info like this one.